assets/scripts/info_bubble/RemoveButton.jsx

```diff
@@ -1,23 +1,16 @@
 import React from 'react'
 import PropTypes from 'prop-types'
-import { FormattedMessage, injectIntl } from 'react-intl'
+import { FormattedMessage, useIntl } from 'react-intl'
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
 import { trackEvent } from '../app/event_tracking'
 import { removeSegment, removeAllSegments } from '../segments/remove'
 import { ICON_TRASH } from '../ui/icons'
 import './RemoveButton.scss'
 
-class RemoveButton extends React.PureComponent {
-  static propTypes = {
-    intl: PropTypes.object.isRequired,
-    segment: PropTypes.number.isRequired
-  }
-
-  static defaultProps = {
-    segment: null
-  }
+const RemoveButton = (props) => {
+  const intl = useIntl()
 
-  onClick = (event) => {
+  const onClick = (event) => {
```

Comment:
While we're here, let's rename `onClick` to `handleClick` to be more consistent with how we name other event handlers.

Reply:
Corrected.